app/routes/titiler/algorithms/alerts.py

```diff
@@ -0,0 +1,142 @@
+        default_end_date, description="end date of alert in YYYY-MM-DD format."
+    )
+
+    alert_confidence: DeforestationAlertConfidence = Field(
+        DeforestationAlertConfidence.low, description="Alert confidence"
+    )
+
+    render_type: RenderType = Field(
+        RenderType.true_color,
+        description="Render true color or encoded pixels",
+    )
+
+    # metadata
+    input_nbands: int = 2
+    output_nbands: int = 4
+    output_dtype: str = "uint8"
+
+    def __call__(self, img: ImageData) -> ImageData:
+        """Decode deforestation and last disturbance alert raster data to
+        RGBA."""
+        date_conf_data = img.data[0]
+
+        self.intensity = img.data[1]
+        self.no_data = img.array.mask[0]
+        self.data_alert_confidence = date_conf_data // 10000
+        self.alert_date = date_conf_data % 10000
+
+        self.mask = self.create_mask()
+
+        if self.render_type == RenderType.true_color:
+            rgb = self.create_true_color_rgb()
+            alpha = self.create_true_color_alpha()
+        else:  # encoded
+            rgb = self.create_encoded_rgb()
+            alpha = self.create_encoded_alpha()
+
+        data = np.vstack([rgb, alpha[np.newaxis, ...]]).astype(self.output_dtype)
+        data = np.ma.MaskedArray(data, mask=False)
+
+        return ImageData(data, assets=img.assets, crs=img.crs, bounds=img.bounds)
+
+    def create_mask(self):
+        start_mask = self.alert_date >= (
+            np.datetime64(self.start_date) - np.datetime64(self.record_start_date)
+        )
+        end_mask = self.alert_date <= (
+            np.datetime64(self.end_date) - np.datetime64(self.record_start_date)
+        )
+
+        confidence_mask = (
+            self.data_alert_confidence
+            >= self.conf_colors[self.alert_confidence].confidence
+        )
+        mask = ~self.no_data * start_mask * end_mask * confidence_mask
+
+        return mask
+
+    def create_true_color_rgb(self):
+        r, g, b = self._rgb_zeros_array()
+
+        for properties in self.conf_colors.values():
+            confidence = properties.confidence
+            colors = properties.colors
+
+            r[self.data_alert_confidence >= confidence] = colors.red
+            g[self.data_alert_confidence >= confidence] = colors.green
+            b[self.data_alert_confidence >= confidence] = colors.blue
+
+        return np.stack([r, g, b], axis=0)
+
+    def create_encoded_rgb(self):
+        r, g, b = self._rgb_zeros_array()
+        r = self.alert_date // 255
+        g = self.alert_date % 255
+        b = (self.data_alert_confidence // 3 + 1) * 100 + self.intensity
+
+        return np.stack([r, g, b], axis=0)
+
+    def create_true_color_alpha(self):
+        alpha = np.where(self.mask, self.intensity * 150, 0)
+        return np.minimum(255, alpha)
+
+    def create_encoded_alpha(self):
+        """To be overridden by alert types that implement it."""
+        return self.create_true_color_alpha()
```

Comment:
I understand why this method is implemented this way. But, do you think it is _likely_ to cause confusion?
Should we log what we're doing here? Basically, "Hey, I know you asked for an encoded alpha, but your specialized alert class doesn't support that." And we return the true color alpha just like you're doing.

Reply:
Good idea. An alternative I considered  was to not provide a default behavior and return rgb instead of rgba if this method isn't provided by the  alert class, but went with this because we expect an intensity layer as input so it makes sense to act on that.